Example/Tests/UserTests.swift

```diff
@@ -177,43 +176,86 @@ class UserTests: QuickSpec {
                 _ = try? user.set(accessToken: "hehe", refreshToken: "hehe", idToken: "hehe")
-                expect(newTokens).toNot(beNil())
+                expect(user.tokens).toNot(beNil())
             }
 
-            it("Should not emit set event if set to same") {
+            it("Should not change tokens if set to same") {
                 let user = User(state: .loggedOut)
-                var newTokens: TokenData?
                 _ = try? user.set(accessToken: "hehe", refreshToken: "hehe", idToken: "hehe")
-                _ = user.didSetTokens.register { newTokens = $0.new }
+                let tokens = user.tokens
+                expect(tokens).toNot(beNil())
                 _ = try? user.set(accessToken: "hehe", refreshToken: "hehe", idToken: "hehe")
-                expect(newTokens).to(beNil())
+                expect(user.tokens) == tokens
             }
 
-            it("Should emit set event if any tokens set differently") {
+            it("Should update tokens if set differently") {
                 let user = User(state: .loggedOut)
-                var newTokens: TokenData?
-                _ = try? user.set(accessToken: "hehe", refreshToken: "hehe", idToken: "hehe", userID: "hehe")
-                _ = user.didSetTokens.register { newTokens = $0.new }
+                try? user.set(accessToken: "hehe", refreshToken: "hehe", idToken: "hehe", userID: "hehe")
+                try? user.set(accessToken: "hehe again")
+                expect(user.tokens).toNot(beNil())
+                expect(user.tokens?.refreshToken) == "hehe"
+                expect(user.tokens?.idToken) == "hehe"
+                expect(user.tokens?.userID) == "hehe"
 
-                _ = try? user.set(accessToken: "hehe again")
-                expect(newTokens).toNot(beNil())
-                newTokens = nil
+                try? user.set(refreshToken: "hehe again")
+                expect(user.tokens).toNot(beNil())
+                expect(user.tokens?.refreshToken) == "hehe again"
 
-                _ = try? user.set(refreshToken: "hehe again")
-                expect(newTokens).toNot(beNil())
-                newTokens = nil
+                try? user.set(idToken: "hehe again")
+                expect(user.tokens).toNot(beNil())
+                expect(user.tokens?.idToken) == "hehe again"
 
-                _ = try? user.set(idToken: "hehe again")
-                expect(newTokens).toNot(beNil())
-                newTokens = nil
+                try? user.set(userID: "hehe again")
+                expect(user.tokens).toNot(beNil())
+                expect(user.tokens?.userID) == "hehe again"
+            }
+        }
 
-                _ = try? user.set(userID: "hehe again")
-                expect(newTokens).toNot(beNil())
+        describe("Keychain management") {
+            it("Should not treat the user as logged in if it does not find keychain data") {
+                let user = User(state: .loggedOut)
```

Comment:
If you specify state loggedOut then User will not even try and set tokens. I guess you're missing a call to `try? user.loadStoredTokens()`?

Reply:
Correct 👍